examples/webpack-demo-vanilla-bundle/src/examples/example07.ts

```diff
@@ -318,69 +320,56 @@ export class Example7 {
 
   allFilters() {
     const grid = this.sgb;
-    const modalHtml: string = `<div id="modal-allFilter" class="modal is-active" >
-        <style type="text/css">
-            #modal-allFilter-table {
-              display: table;
-            }
-
-            #modal-allFilter-table .row {
-                display: table-row;
-            }
-
-            #modal-allFilter-table .column {
-                display: table-cell;
-                vertical-align: top;
-                width: 40%;
-            }
-        </style>
-        <div class="modal-background"></div>
-        <div class="modal-card">
-            <header class="modal-card-head">
-                <p class="modal-card-title">Filter</p>
-                <button class="delete btn-close" aria-label="close"></button>
-            </header>
-            <section class="modal-card-body">
-              <div class="slickgrid-container grid-pane">
-                <div id="modal-allFilter-content">
-                  <div id="modal-allFilter-table" class="slick-headerrow ui-state-default">
-                  </div>
-                </div>
+    const modalHtml = `<div id="modal-allFilter" class="modal is-active">
+      <div class="modal-background"></div>
+      <div class="modal-card">
+        <header class="modal-card-head">
+          <p class="modal-card-title">Filter</p>
+          <button class="delete btn-close" aria-label="close"></button>
+        </header>
+        <section class="modal-card-body">
+          <div class="slickgrid-container grid-pane">
+            <div id="modal-allFilter-content">
+              <div id="modal-allFilter-table" class="slick-headerrow ui-state-default">
               </div>
-            </section>
-            <footer class="modal-card-foot">
-              <button class="button btn-close">Close</button>
-              <button id="btn-clear-all" class="button">Clear All Filter</button>
-              <button class="button btn-close is-success">Search</button>
-            </footer>
-        </div>
+            </div>
+          </div>
+        </section>
+        <footer class="modal-card-foot">
+          <button class="button btn-close">Close</button>
+          <button id="btn-clear-all" class="button">Clear All Filter</button>
+          <button class="button btn-close is-success">Search</button>
+        </footer>
+      </div>
     </div>`;
 
-    document.body.appendChild(DOMPurify.sanitize(modalHtml, { RETURN_DOM: true }));
```

Comment:
@mcallegario I think you used `RETURN_DOM` because `appendChild` didn't accept the element otherwise but please note that it was causing issues and I had to remove that and I had to revert to using jQuery instead for appending to the real body, see animated gif included

Reply:
Oops! I'm very sorry!
I didn't know (and didn't even imagine) that DOMPurify adds an extra BODY to the html document.

Reply:
I use slickgrid-universal on a page within Sharepoint. I started having problem designing the selection filter menu after these code changes were made:
![image](https://user-images.githubusercontent.com/44271732/146612504-2f7da520-de0c-4b66-b23b-2a802e012639.png)

To work 100% and I have to always comment out these lines(170,226) which calls the method "subscribeToOnHeaderRowCellRendered".

Otherwise, it starts to happen:
![image](https://user-images.githubusercontent.com/44271732/146612535-d61a852a-ea46-4996-935c-24db28f512fa.png)

Reply:
that code is there for a reason, to delete all filters and avoid leaks, it's not a good idea to remove it and is meant to destroy/remove the filter. Did you really remove the DOM Sanitize `RETURN_DOM`? I can't really help with a repro

Reply:
So, in the original version, this problem occurred after the commit of 08-29-2020 "[fix(filters): remove Filters from DOM after header row gets destroyed](https://github.com/ghiscoding/slickgrid-universal/commit/b08d4ba070ec9d9d131d6830e4625e6ef950ac09)"
But don't worry, I believe it's something specific to my Sharepoint environment (webpart) with vanilla version, unfortunately I didn't have free time to investigate the problem further (debug)

Reply:
that code change was because Filters were not being removed, they became orphans in the DOM. Every time I was clicking on a select dropdown filter, it was creating a new dropdown filter without removing it from the DOM after closing it, so the more I click the filter, the more orphan dropdown were being created, so it was a big memory leak, not visible to the user but visible in the DOM. The Multiple/Single Filter is special, it create and append the dropdown to the `body` every time you open it and when you close the dropdown, it destroys it (removes it) from the DOM, in your use case it doesn't seem to find the dropdown (perhaps because of the latest filter PR you made, I'm not sure)